File: app/src/main/java/com/loopeer/codereader/model/DirectoryNode.java
package com.loopeer.codereader.model;

import java.util.List;

public class DirectoryNode extends BaseModel implements Comparable{

    public String name;
    public List<DirectoryNode> pathNodes;
    public boolean isDirectory;
    public boolean openChild;
    public int depth;
    public String displayName;

    public DirectoryNode() {
    }

    public DirectoryNode(String name) {
        this.name = name;
    }

    public DirectoryNode(List<DirectoryNode> pathNodes, String name) {
        this.pathNodes = pathNodes;
        this.name = name;
    }

    @Override
    public int compareTo(Object o) {
        if (!(o instanceof DirectoryNode)) {
            return 1;
        }
        if (((DirectoryNode) o).isDirectory && !isDirectory) {
            return 1;
        }
        if (!((DirectoryNode) o).isDirectory && isDirectory) {
            return -1;
        }
        return name.compareTo(((DirectoryNode) o).name);
    }
}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/model/DirectoryNode.kt
package com.loopeer.codereaderkt.model


class DirectoryNode{

    lateinit var name: String
    lateinit var pathNodes: List<DirectoryNode>
    var isDirectory: Boolean = false
    var openChild: Boolean = false
    var depth: Int = 0
    var displayName: String? = null
    var absolutePath: String? = null

    constructor(name: String) {
        this.name = name
    }

    constructor(pathNodes: List<DirectoryNode>, name: String) {
        this.pathNodes = pathNodes
        this.name = name
    }

    operator fun compareTo(o: Any): Int {
        if (o !is DirectoryNode) {
            return 1
        }
        if (o.isDirectory && !isDirectory) {
            return 1
        }
        if (!o.isDirectory && isDirectory) {
            return -1
        }
        return name.compareTo(o.name)
    }

    constructor()

}